feat: add health check endpoint for principal

jparsai opened the pull request:
This PR is to add health check endpoint to principal, which can be used to verify server readiness. This would fix the issue https://github.com/argoproj-labs/argocd-agent/issues/422, where principal is not fully started but tests still executed which causes test failure.
This PR also exposes health check via a service for both Principal and Agent.

jannfis commented on test/e2e2/cache_test.go:
```diff
@@ -28,19 +28,45 @@ import (
 
+const (
+	PrincipalName     = "principal"
+	AgentName_Managed = "agent-managed"
```
Please do not use underscores in symbol naming (https://go.dev/doc/effective_go#mixed-caps)

jannfis reviewed (commented):
Thanks, I only have a minor nit otherwise LGTM.

We can also expose this as a service, so that the cluster can make use of the health check too.

jparsai commented:
> Thanks, I only have a minor nit otherwise LGTM.
> 
> We can also expose this as a service, so that the cluster can make use of the health check too.

Thanks Jann, I added service for it. Since health check is also available in agent, I added service for agent as well. PTAL.

jannfis reviewed (approved):
LGTM, thanks!
